contrib/win32/win32compat/w32-sshfileperm.c

```diff
@@ -164,3 +164,69 @@ check_secure_file_permission(const char *input_path, struct passwd * pw)
 	return ret;
 }
 
+#ifdef WINDOWS
+char*
+get_execpath(const char **av)
+{
+	if (NULL == av) {
+		errno = EINVAL;
+		return NULL;
+	}
+
+	char* retVal = NULL;
+	const char *prgPath = av[0];
+	char* exePath = _strdup(prgPath);
+	char* exePathPtr = exePath;
+
+	if ('/' == exePathPtr[0] || '\\' == exePathPtr[0]) {
+		++exePathPtr;
+	}
+	convertToBackslash(exePathPtr);
+
+	/*
+	* check if av[0] starts with 'C:\Program Files\PowerShell' or 'C:\Windows\System32\WindowsPowerShell'
```

Comment:
Why check specific paths? Windows Powershell is one thing but pwsh could be installed to another  drive, this needs to check PATH for the pwsh executable and in implementing that, would be easier to use it for Windows Powershell too.